### ISSUE TITLE
Spectating players break when target gets disconnected

### ISSUE THREAD

**Issue by xunder-matth** [openmultiplayer/open.mp#735]
Spectating players will break when spectating target gets disconnected. 

To reproduce this you need 3 players connected. 
Commands I've used:
```
CMD:specon(playerid, params[]) {
    if (IsNull(params)) {
        return SendClientMessage(playerid, -1, "/specon [target id]");
    }

    new targetid = strval(params);

    if (!IsPlayerConnected(targetid)) {
        return SendClientMessage(playerid, -1, "Invalid target id");
    }

    SetPlayerInterior(playerid, GetPlayerInterior(targetid));
	SetPlayerVirtualWorld(playerid, GetPlayerVirtualWorld(targetid));

	TogglePlayerSpectating(playerid, true);

	if (IsPlayerInAnyVehicle(targetid)) {
		PlayerSpectateVehicle(playerid, GetPlayerVehicleID(targetid));
	} else {
		PlayerSpectatePlayer(playerid, targetid);
	}
    return 1;
}

CMD:specoff(playerid, params[]) {
    TogglePlayerSpectating(playerid, false);
    return 1;
}
```

1. Get 3 players on server (let's say ids are 0, 1 and 2)
2. Player id 0: /specon 2 - this should be successful
3. Disconnect player id 2 - this should cancel player's id 0 spectating state
4. Try spectating player id 1 using player id 0 - this should fail

This bug is presented in [v1-RC2](https://github.com/openmultiplayer/open.mp/commit/dcd0190a91e67c99217e8ed3254baeb86733a0e4) and [7b22bb036aafda001733649f67f5119ebf18b0cf](https://github.com/openmultiplayer/open.mp/commit/7b22bb036aafda001733649f67f5119ebf18b0cf)

Bug is not presented in SA-MP server.

**Comment by PazzOnee** [comment]
**Also to note, when do TogglePlayerSpectatig spectator to false, you again can spectate normaly**

**Comment by AmyrAhmady** [comment]
Fixed in #737

**Comment by PazzOnee** [comment]
@AmyrAhmady Still not fixed in #737

**Comment by AmyrAhmady** [comment]
Must now be fixed in #740
